.github/workflows/main.yml

```diff
@@ -359,6 +344,104 @@ jobs:
           parallel: true
           path-to-lcov: ./coverage.lcov
 
+  notebook-matrix:
+    runs-on: ubuntu-latest
+    needs:
+      [
+        check-changes,
+        prepare-nonhindcast-cache,
+        prepare-wave-hindcast-cache,
+        prepare-wind-hindcast-cache,
+      ]
+    if: |
+      always() &&
+      (
+        (
+          needs.prepare-nonhindcast-cache.result == 'success' &&
+          needs.prepare-wave-hindcast-cache.result == 'skipped' &&
+          needs.prepare-wind-hindcast-cache.result == 'skipped' &&
+          needs.check-changes.outputs.should-run-hindcast == 'false'    
+        ) ||
+        (
+          needs.prepare-nonhindcast-cache.result == 'success' &&
+          needs.prepare-wave-hindcast-cache.result == 'success' &&
+          needs.prepare-wind-hindcast-cache.result == 'success' &&
+          needs.check-changes.outputs.should-run-hindcast == 'true' 
+        )
+      )
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11'
+
+      - name: Generate matrix
+        id: set-matrix
+        run: |
+          matrix_json=$(python .github/workflows/generate_notebook_matrix.py)
```

Comment:
What is the purpose of the ``matrix_json`` variable here? Isn't ``matrix`` the only variable from ``generate_notebook_matrix.py`` that is used? Similarly, where is the ``matrix`` key "include" being used?

Reply:
In bash here we are setting a variable `matrix_json` 
On the next line we echo that variable to send it as std_out via ">>" to the variable `matrix`.

So the `matrix_json` variable is used in the echo command